specs/governance/gov-token.md

```diff
@@ -0,0 +1,93 @@
+| Token symbol      | `OP`                                         |
+
+`GovernanceToken` is an [ERC20](https://eips.ethereum.org/EIPS/eip-20) token contract that inherits from `ERC20Burnable`,
+`ERC20Votes`, and `Ownable`. It allows token holders to delegate their voting power to other addresses, enabling a representative
+voting system.
+
+The specific implementation details of the voting and delegation logic are inherited from the `ERC20Votes` contract. The
+`GovernanceToken` contract focuses on integrating this functionality with the ERC20 token standard and adding the minting
+capability.
+
+### Token Minting
+
+The contract MUST have a `mint` function with external visibility that allows the contract owner to mint new tokens to an
+arbitrary address. This function MUST only be called by the contract owner, as enforced by the `onlyOwner` modifier inherited
+from the `Ownable` contract. When tokens are minted, the voting power of the recipient address MUST be updated accordingly.
+
+### Token Burning
+
+The contract MUST allow token holders to burn their own tokens using the inherited `burn` function from `ERC20Burnable`.
+When tokens are burned, the total supply and the holder's voting power MUST be reduced accordingly.
+
+### Voting Power
+
+Each token corresponds to one unit of voting power.
+By default, token balance does not account for voting power. To have their voting power counted, token holders MUST delegate
+their voting power to an address (can be their own address).
+The contract MUST offer public accessors for quering voting power, as outlined below.
+
+#### Queries
+
+- The `getVotes` function MUST return the current voting power of an address.
+- The `getPastVotes` function MUST allow querying the voting power of an address at a specific block number in the past.
+- The `getPastTotalSupply` function MUST return the total voting power at a specific block number in the past.
+
+### Delegation
+
+Vote power can be delegated either by calling the `delegate` function directly or by providing a signature to be used
+with `delegateBySig`, as inherited from `ERC20Votes`.
+
+When a token holder delegates their voting power, the delegated address receives the voting power corresponding to the token
+holder's balance.
+
+Delegation is recorded in checkpoints, which store the voting power of each address at specific points in time.
+
+### Checkpoints
+
+The contract keeps a history of each account's voting power using checkpoints.
+Checkpoints are recorded whenever tokens are transferred, minted, or burned, and when delegation occurs.
+Each checkpoint represents the voting power of an address at a specific block number.
+The contract MUST maintain a mapping of addresses to their checkpoint history. Additionally, the contract MUST
+provide functions `numCheckpoints` and `checkpoints` to allow retrieving the number of checkpoints for an account
+and the details of a specific checkpoint.
+
+### Overrides
```

Comment:
I think that you can delete this section as its very implementation specific and not really important about the way that it works. You could imagine its written in a different language that doesn't have overrides